### PR TITLE
Add eye dropper

### PR DESCRIPTION
**Pull request by KryptKode**
**Notes**
Add eye dropper

https://user-images.githubusercontent.com/25648077/147536895-8ac4cccb-b982-4eaa-bf08-f8c7fb8da90b.mov

### CODE REVIEW

File: app/src/main/kotlin/com/simplemobiletools/draw/pro/helpers/EyeDropper.kt [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+    private fun getPixelAtPoint(bitmap: Bitmap, x: Int, y: Int): Int {
+        if (bitmap.isValidCoordinate(x, y)) {
+            return bitmap.getPixel(x, y)
+        }
+        return NO_COLOR
+    }
+
+    private fun Bitmap.isValidCoordinate(x: Int, y: Int): Boolean {
+        return x in 1 until width && y in 1 until height
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/draw/pro/helpers/EyeDropper.kt:81]: simplify this please, looks too confusing to read

Comment (KryptKode) [reply app/src/main/kotlin/com/simplemobiletools/draw/pro/helpers/EyeDropper.kt]: Done. Please check @tibbi

File: app/src/main/kotlin/com/simplemobiletools/draw/pro/activities/MainActivity.kt [hunk 2]
```diff
@@ -60,10 +61,17 @@ class MainActivity : SimpleActivity(), CanvasListener {
     private var savedPathsHash = 0L
     private var lastSavePromptTS = 0L
     private var isEraserOn = false
+    private var isEyeDropperOn = false
     private var isImageCaptureIntent = false
     private var isEditIntent = false
     private var lastBitmapPath = ""
 
+    private val eyeDropper by lazy {
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/draw/pro/activities/MainActivity.kt:69]: we dont use lazy in any app, please write it the casual way

File: app/src/main/kotlin/com/simplemobiletools/draw/pro/activities/MainActivity.kt [hunk 3]
```diff
@@ -335,6 +345,16 @@ class MainActivity : SimpleActivity(), CanvasListener {
...
     }
 
+    private fun eyeDropperClicked() {
+        isEyeDropperOn = !isEyeDropperOn
+        if (isEyeDropperOn) {
+            eyeDropper.start()
+        } else {
+            eyeDropper.stop()
+        }
+        eye_dropper.setImageDrawable(ContextCompat.getDrawable(this, if (isEyeDropperOn) R.drawable.ic_colorise_off_vector else R.drawable.ic_colorize_vector))
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/draw/pro/activities/MainActivity.kt:355]: it is "colorize", not "colorise". Also move the if/else into some variable, this like has way too many things going on

**Comment by tibbi** [comment]
added a quick small notes, it seems to work just fine though. Also please show R.string.eyedropper on long pressing it

**Comment by tibbi** [comment]
should be good, thanks

**Pull request merged**